### PR TITLE
robust plot documentation

### PR DESCRIPTION
**Pull request by clarkfitzg**
These are the plots that will be in the docs.

## default

image

## `robust=True`

image

### CODE REVIEW

File: xray/plot/plot.py [hunk 1]
```diff
@@ -211,7 +211,8 @@ def _update_axes_limits(ax, xincrease, yincrease):
 
 def _determine_cmap_params(plot_data, vmin=None, vmax=None, cmap=None,
                            center=None, robust=False, extend=None,
-                           levels=None, filled=True, cnorm=None):
+                           levels=None, filled=True, cnorm=None,
+                           robust_percentile=2):
```
Comment (jhamman) [inline xray/plot/plot.py]: do we really need both the `robust` and `robust_percetile` keyword arguments.  Maybe just allow `robust` to be `None`, `True`, `False`, or an integer specifying the percentile (default is None / False and True defaults to 2).

File: xray/plot/plot.py [hunk 2]
```diff
@@ -392,7 +393,7 @@ def _plot2d(plotfunc):
         use of a diverging colormap.
     robust : bool, optional
         If True and ``vmin`` or ``vmax`` are absent, the colormap range is
-        computed with robust quantiles instead of the extreme values.
+        computed with 2nd and 98th percentiles instead of the extreme values.
```
Comment (jhamman) [inline xray/plot/plot.py]: With the `robust_percentile`, this is not always true.

File: xray/plot/plot.py [hunk 3]
```diff
@@ -444,7 +445,8 @@ def newplotfunc(darray, ax=None, xincrease=None, yincrease=None,
         filled = plotfunc.__name__ != 'contour'
 
         cmap_params = _determine_cmap_params(z.data, vmin, vmax, cmap, center,
-                                             robust, extend, levels, filled)
+                                             robust, extend, levels,
+                                             filled, robust_percentile=2)
```
Comment (jhamman) [inline xray/plot/plot.py]: shouldn't this come in as a kwarg?

File: xray/plot/plot.py (hunk 2 above)
Comment (clarkfitzg) [inline xray/plot/plot.py]: @jhamman the thought here is to hide complexity from the user by only having them see `robust=True` or `False`. If they want to do anything beyond this they can just do the operation on a `DataArray`.

File: xray/plot/plot.py (hunk 3 above)
Comment (clarkfitzg) [inline xray/plot/plot.py]: see above

**Comment by clarkfitzg** [comment]
Was trying to keep the internal code a little more modular by parameterizing the percentiles. 

But it was simpler before. Maybe should keep it like it was.

**Comment by clarkfitzg** [comment]
@jhamman I took out the `robust_percentile` arg. You're right, it's better this way.

**Comment by jhamman** [comment]
I wasn't necessarily against being able to set the percentiles used, I just didn't think we needed two args to do it.

**Comment by clarkfitzg** [comment]
@jhamman any more feedback before merge?

**Comment by jhamman** [comment]
Nope.  As long as the docs build, this is good to go.

**Comment by clarkfitzg** [comment]
Merging

**Comment by shoyer** [comment]
well done guys!

**Pull request merged**